packages/helix/project.bri

```diff
@@ -0,0 +1,48 @@
+import * as std from "std";
+import git from "git";
+import rust, { cargoBuild } from "rust";
+
+export const project = {
+  name: "helix",
+  version: "25.01.1",
+  repository: "https://github.com/helix-editor/helix.git",
+};
+
+const source = Brioche.gitCheckout({
+  repository: project.repository,
+  ref: project.version,
+});
+
+export default function helix(): std.Recipe<std.Directory> {
+  return cargoBuild({
+    env: {
+      HELIX_DISABLE_AUTO_GRAMMAR_BUILD: "1",
+    },
+    path: "helix-term",
+    runnable: "hx",
+    source,
+  });
```

Comment:
There is a way to build the tree sitter grammar, you could remove your env var `HELIX_DISABLE_AUTO_GRAMMAR_BUILD`, and include `git` in the dependencies + enabling network access during the building. This way the grammars will be pulled, and built:


```suggestion
  return cargoBuild({
    source,
    path: "helix-term",
    dependencies: [git],
    runnable: "bin/hx",
    unsafe: { networking: true },
  });
```

Reply:
yeah, i wasn't sure whether we were leaning towards 100% hermetic builds

there's probably a way to seal the files required for the grammar to build ...

Reply:
I first thought that retrieving the package source from a GitHub tarball as pointed it out [here](https://github.com/NixOS/nixpkgs/blob/32a4e87942101f1c9f9865e04dc3ddb175f5f32e/pkgs/by-name/he/helix/package.nix#L15) would be enough, but when I quickly tried I wasn't able to build Helix entirely. Maybe I misconfigured something, I haven't further dig.

Reply:
Personally, I am not super picky about perfect hermeticity! I haven't looked super closely at how Helix is downloading languages, but first impressions is it looks like it's pretty stable? If so, it seems like we don't lose much by just enabling an unsafe build (we do lose caching if any of the upstream URLs are unavailable), and just using unsafe would make it more maintainable, since we don't need to deal with their config file manually. Then again, it seems like it's not a _lot_ of work to parse and download?

...anyway, I'm pretty split either way, I think either just allowing unsafe downloads or manually parsing would be fine